src/extension.ts

```diff
@@ -48,45 +48,71 @@ async function fetchPackageInfo(requirement: PackageRequirement): Promise<Packag
     try {
         const response: Response = await fetch(`https://pypi.org/pypi/${requirement.id}/json`)
         let info: PackageInfo | null = null
-        if (response.ok) info = (await response.json()).info
+        if (response.ok) info = (await response.json() as { info: PackageInfo}).info
         return [response.status, info]
     } catch (e) {
         return [null, null]
     }
 }
 
-function presentPackageInfo(info: PackageInfo): string {
+function presentPackageInfo(info: PackageInfo): string[] {
     const summarySubPart: string = info.summary ? ` – ${linkify(info.summary, info.home_page)}` : ''
     const headPart: string = `**${linkify(info.name, info.package_url)}${summarySubPart}**`
     const emailSubpart: string = info.author_email ? ` (${info.author_email})` : ''
     const authorSubpart: string = info.author && info.author_email ? `By ${info.author}${emailSubpart}.` : ''
     const licenseSubpart: string = info.license ? ` ${info.license.replace(/ licen[cs]e/gi, '')} licensed.` : ''
     const versionPart: string = `Latest version: ${linkify(info.version, info.release_url)}.`
-    const infoPresentation: string = [
-        headPart, authorSubpart + licenseSubpart, versionPart
-    ].filter(Boolean).join('\n\n')
+    const versionPartLens: string = `Latest version: ${info.version}`
+    const infoPresentation: Array<string> = [
+        headPart, authorSubpart + licenseSubpart, versionPart, versionPartLens
+    ].filter(Boolean)
+
     return infoPresentation
 }
 
 async function provideHover(document: vscode.TextDocument, position: vscode.Position) {
     const requirement: PackageRequirement | null = extractPackageRequirement(document.lineAt(position.line))
     if (requirement === null) return new vscode.Hover('')
-    let infoPresentation: string | undefined = infoPresentationCache.get(requirement.id)
-    if (infoPresentation === undefined) {
-        const [status, info]: PackageInfoRequest = await fetchPackageInfo(requirement)
-        if (status === 200) infoPresentation = presentPackageInfo(info!)
-        else if (status === 404) infoPresentation = `{id_raw} is not available in PyPI`
-        if (infoPresentation) infoPresentationCache.set(requirement.id, infoPresentation)
-        else infoPresentation = linkify(
-            `could not fetch ${requirement.id_raw} information from PyPI`,
-            `https://pypi.org/project/${requirement.id_raw}/`
-        )
-    }
-    infoPresentation = infoPresentation.replace('{id_raw}', requirement.id_raw)
```

Comment:
Why was this problem handling removed?

Reply:
Transferred to the CodeLens provider below.